lbrynet/cryptstream/CryptBlob.py

```diff
@@ -67,13 +70,17 @@ def finish_decrypt():
             last_chunk = self.cipher.update(data_to_decrypt) + self.cipher.finalize()
             write_func(remove_padding(last_chunk))
 
-        def decrypt_bytes(data):
+
+        read_handle = self.blob.open_for_reading()
+
+        def decrypt_bytes():
+            data = read_handle.read()
             self.buff += data
             self.len_read += len(data)
             write_bytes()
+            finish_decrypt()
 
-        d = self.blob.read(decrypt_bytes)
-        d.addCallback(lambda _: finish_decrypt())
+        d = threads.deferToThread(decrypt_bytes)
```

Comment:
Why deferToThread?

Reply:
Before this PR StreamBlobDecryptor used FileSender() and the producer/consumer pattern to decrypt blobs in an asynchronous manner. 

If we dont't deferToThread, this will block the reactor till a standard read()  and write() to a normal file handle is finished ( the write() is called by write_bytes() see lbry_file/client/EncryptedFileDownloader.py EncryptedFileSaver._get_write_func() )

Reply:
I think it would be better to pass the read_handle to a FileBodyProducer and call decrypt_bytes on the result of that instead of doing the read() in the decrypt_bytes function